core/src/main/java/feast/core/model/Job.java

```diff
@@ -117,4 +118,25 @@ public String getSinkName() {
 
     return ingestJob;
   }
+
+  @Override
+  public int hashCode() {
+    return Objects.hash(this.source, this.store, this.runner);
+  }
+
+  @Override
+  public boolean equals(Object obj) {
+    if (this == obj) return true;
+    if (!super.equals(obj)) return false;
+    if (getClass() != obj.getClass()) return false;
+    Job other = (Job) obj;
+    if (!runner.equals(other.runner)) {
```

Comment:
since there's hash implementation - you can reuse it

Reply:
I don't think that would work. `Object.hash()` by definition could return a collision, so `.equals()` might not work as expected for such collisions when they happen. Direct comparison in `.equals()` ensure that works as expected for all cases.